packages/platform-server/src/transfer_state.ts

```diff
@@ -17,11 +18,20 @@ import {
 
 import {BEFORE_APP_SERIALIZED} from './tokens';
 
+// Tracks whether an app with given ID has been rendered.
+export const TRANSFER_STATE_SERIALIZED_FOR_APPID = new InjectionToken<Set<string>>(
+  typeof ngDevMode === 'undefined' || ngDevMode ? 'TRANSFER_STATE_SERIALIZED_FOR_APPID' : '',
+  {
+    providedIn: 'platform',
+    factory: () => new Set(),
+  },
+);
+
 export const TRANSFER_STATE_SERIALIZATION_PROVIDERS: Provider[] = [
   {
     provide: BEFORE_APP_SERIALIZED,
     useFactory: serializeTransferStateFactory,
-    deps: [DOCUMENT, APP_ID, TransferState],
+    deps: [DOCUMENT, APP_ID, TransferState, TRANSFER_STATE_SERIALIZED_FOR_APPID],
```

Comment:
This would make the `TRANSFER_STATE_SERIALIZED_FOR_APPID` non-tree-shakable, unfortunately. To resolve this, we can inject an `Injector` and inject particular tokens inside of the `serializeTransferStateFactory` as needed (see also a comment about this below).
```suggestion
    deps: [Injector],
```

Reply:
Alternatively, we can also inject the `Injector` within the `serializeTransferStateFactory` function itself via `inject(Injector)` call.